raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansProcessor.java

```diff
@@ -196,7 +242,101 @@ public void process(Record<TraceIdentity, RawSpan> record) {
+    Optional<String> maybePeerPort = HttpSemanticConventionUtils.getPeerPort(event);
+    String serviceName = event.getServiceName();
+    PeerIdentity peerIdentity =
+        PeerIdentity.newBuilder()
+            .setIpIdentity(
+                IpIdentity.newBuilder()
+                    .setTenantId(tenantId)
+                    .setEnvironment(environment)
+                    .setHostAddr(maybeHostAddr.orElse(null))
+                    .setPeerAddr(maybePeerAddr.orElse(null))
+                    .setPeerPort(maybePeerPort.orElse(null))
+                    .build())
+            .build();
+    if (PeerIdentityValidator.isValid(peerIdentity)) {
+      this.peerIdentityToSpanMetadataStateStore.put(
+          peerIdentity,
+          SpanMetadata.newBuilder()
+              .setServiceName(serviceName)
+              .setEventId(HexUtils.getHex(event.getEventId()))
+              .build());
+    }
+  }
+
+  // get the service name for that peer service identity and correlate the current span
+  private void handleServerSpan(String tenantId, Event event, String environment) {
+    Optional<String> maybePeerAddr = HttpSemanticConventionUtils.getPeerIpAddress(event);
+    Optional<String> maybeHostAddr = HttpSemanticConventionUtils.getHostIpAddress(event);
+    Optional<String> maybeHostPort = HttpSemanticConventionUtils.getHostPort(event);
+    PeerIdentity peerIdentity =
+        PeerIdentity.newBuilder()
+            .setIpIdentity(
+                IpIdentity.newBuilder()
+                    .setTenantId(tenantId)
+                    .setEnvironment(environment)
+                    .setHostAddr(maybePeerAddr.orElse(null))
+                    .setPeerAddr(maybeHostAddr.orElse(null))
+                    .setPeerPort(maybeHostPort.orElse(null))
+                    .build())
+            .build();
+    if (PeerIdentityValidator.isValid(peerIdentity)) {
+      SpanMetadata spanMetadata = this.peerIdentityToSpanMetadataStateStore.get(peerIdentity);
+      if (Objects.nonNull(spanMetadata)) {
+        logger.debug(
+            "Adding {} as: {} from spanId: {} in spanId: {} with service name: {}",
+            PEER_SERVICE_NAME,
+            spanMetadata.getServiceName(),
+            spanMetadata.getEventId(),
+            HexUtils.getHex(event.getEventId()),
+            event.getServiceName());
+
+        event
+            .getEnrichedAttributes()
+            .getAttributeMap()
+            .put(PEER_SERVICE_NAME, AttributeValueCreator.create(spanMetadata.getServiceName()));
+      }
+    }
+  }
+
+  private boolean isPeerServiceNameIdentificationRequired(Event event) {
+    String agentType =
+        SpanAttributeUtils.getStringAttributeWithDefault(
+            event, this.peerCorrelationAgentTypeAttribute, null);
+    return Objects.nonNull(agentType)
+        && this.peerCorrelationEnabledAgents.contains(agentType)
+        && (this.peerCorrelationEnabledCustomers.contains(event.getCustomerId())
```

Comment:
The tenant condition should come first in the `order`. If we have not enabled for a given tenant, don't do anything.

We don't have to extract the stringAttribute if its not enabled for customer.

Reply:
Add template config - `enabled.all.customer = 'true` or `false` - instead of code level constant.